account_commission/models/commission_settlement.py

```diff
@@ -159,6 +159,47 @@ def make_invoices(self, journal, product, date=False, grouped=False):
         self.write({"state": "invoiced"})
         return invoices
 
+    def grouped_report_lines(self):
+        self.ensure_one()
+        if self.settlement_type == "sale_invoice":
+            grouped_res = {}
+            for line in self.line_ids:
+                origin = (
+                    line.invoice_line_id.move_id if line.invoice_line_id else "Manual"
+                )
+                if (line.date, origin) not in grouped_res:
+                    grouped_res[line.date, origin] = {
+                        "commission": 0.0,
+                        "base_commission": 0.0,
+                    }
+                grouped_res[line.date, origin]["commission"] += line.settled_amount
+                if line.move_id:
+                    sign = 1
+                    if "refund" in line.move_id.move_type:
+                        sign = -1
+                    grouped_res[line.date, origin]["base_commission"] += (
+                        sign * line.invoice_line_id.price_subtotal
+                    )
+
+            # prettier for report
+            res = []
+            for key1, key2 in grouped_res.keys():
+                res.append(
+                    {
+                        "date": key1,
+                        "settled_amount": grouped_res[key1, key2]["commission"],
+                        "base_commission": grouped_res[key1, key2]["base_commission"],
+                        "commision": "",
```

Comment:
Wouldn't it be better to delete this if it's empty? Or perhaps the commission name can be displayed if there is only one type